backend/index.js

```diff
@@ -1,20 +1,52 @@
 const AWS = require('aws-sdk');
+const needle = require('needle');
 const dynamo = new AWS.DynamoDB.DocumentClient();
 
-/** Endpoint: returns song recommendations */
+/* Endpoints */
+/** healthCheck: Simply returns "healthy" as a basic server health check. */
+healthCheck = function(data) {
+  return ['Healthy!', 200];
+};
+/** findSong: Searches for song in the Spotify API. */
+/* Parameters:
+ * query(string): What the user is searching
+ * type(string): Album, artist, playlist, track, show, or episode
+ * OPTIONAL - market(string): Localization. Default is US
+ * OPTIONAL - limit(int): How many entries to return. Default is 1
+ */
+/* Reference: https://developer.spotify.com/documentation/web-api/reference/#endpoint-search */
 findSong = function(data) {
-  return ['pepperoni pizza\n', 200];
+  // build our API string
+  const baseUrl = 'https://api.spotify.com/v1/search?q=';
+  const query = data.query;
+  const type = data.type;
+  const builtUrl = (baseUrl + query + '&type=' + type).replace(' ', '%20');
+  // API call
+  const options = {
+    headers: {
+      'content-type': 'application/json',
+      'authorization': 'Bearer BQAGw3MEt46ILH1rqAwnwPyFnugROjI3hwaPD9sJX_RywsvsNQ7nodr7UXzTp617eZdpn5fxMTg3NZLF-KkN1bIGKOg_gmaAPxeXdi3X8CbknvnCRPPvXagbDeoFCjwE8S0vakqJq9BZoK1LuqM',
+    },
+  };
+  needle.request('get', builtUrl, options, function(error, response) {
+    if (!error && response.statusCode == 200) {
+      return response.body;
```

Comment:
these return statements return for the callback, not the outer function unfortunately. We'll need to redo these with promises and await statements.